meson.build

```diff
@@ -0,0 +1,629 @@
+    'src/tagmanager/tm_tag.h',
+    'src/tagmanager/tm_tag.c',
+    'src/tagmanager/tm_workspace.h',
+    'src/tagmanager/tm_workspace.c',
+    c_args: geany_cflags,
+    dependencies: [dep_ctags, glib])
+dep_tagmanager = declare_dependency(link_with: tagmanager, include_directories: [itagmanager])
+
+# Generate signallist.i
+sh = find_program('sh')
+gen_src = custom_target('gen-signallist',
+                        input : [ 'scripts/gen-signallist.sh', 'data/geany.glade' ],
+                        output : [ 'signallist.i' ],
+                        command : [sh, '@INPUT0@', '@INPUT1@', '@OUTPUT@' ])
+
+libgeany = shared_library('geany',
+    'src/about.c',
+    'src/about.h',
+    'src/app.h',
+    'src/build.c',
+    'src/build.h',
+    'src/callbacks.c',
+    'src/callbacks.h',
+    'src/dialogs.c',
+    'src/dialogs.h',
+    'src/document.c',
+    'src/document.h',
+    'src/editor.c',
+    'src/editor.h',
+    'src/encodings.c',
+    'src/encodings.h',
+    'src/filetypes.c',
+    'src/filetypes.h',
+    'src/geanyentryaction.c',
+    'src/geanyentryaction.h',
+    'src/geanymenubuttonaction.c',
+    'src/geanymenubuttonaction.h',
+    'src/geanyobject.c',
+    'src/geanyobject.h',
+    'src/geanywraplabel.c',
+    'src/geanywraplabel.h',
+    'src/gtkcompat.h',
+    'src/highlighting.c',
+    'src/highlighting.h',
+    'src/highlightingmappings.h',
+    'src/keybindings.c',
+    'src/keybindings.h',
+    'src/keyfile.c',
+    'src/keyfile.h',
+    'src/log.c',
+    'src/log.h',
+    'src/libmain.c',
+    'src/main.h',
+    'src/geany.h',
+    'src/msgwindow.c',
+    'src/msgwindow.h',
+    'src/navqueue.c',
+    'src/navqueue.h',
+    'src/notebook.c',
+    'src/notebook.h',
+    'src/plugins.c',
+    'src/plugins.h',
+    'src/pluginutils.c',
+    'src/pluginutils.h',
+    'src/prefs.c',
+    'src/prefs.h',
+    'src/printing.c',
+    'src/printing.h',
+    'src/project.c',
+    'src/project.h',
+    'src/sciwrappers.c',
+    'src/sciwrappers.h',
+    'src/search.c',
+    'src/search.h',
+    'src/socket.c',
+    'src/socket.h',
+    'src/spawn.c',
+    'src/spawn.h',
+    'src/stash.c',
+    'src/stash.h',
+    'src/support.h',
+    'src/symbols.c',
+    'src/symbols.h',
+    'src/templates.c',
+    'src/templates.h',
+    'src/toolbar.c',
+    'src/toolbar.h',
+    'src/tools.c',
+    'src/tools.h',
+    'src/sidebar.c',
+    'src/sidebar.h',
+    'src/ui_utils.c',
+    'src/ui_utils.h',
+    'src/utils.c',
+    'src/utils.h',
+    'src/vte.c',
+    'src/vte.h',
+    gen_src,
+    soversion: '0',
+    c_args: geany_cflags,
+    include_directories: [iscintilla],
+    dependencies: [dep_tagmanager, dep_ctags, dep_scintilla] + deps,
+    install: true)
+
+executable('geany', 'src/main.c',
+            link_with: libgeany,
+            c_args: geany_cflags,
+            dependencies: [dep_ctags] + deps,
+            build_rpath: meson.build_root(),
+            install_rpath: join_paths(prefix, 'lib'),
```

Comment:
@kugel- here is why rpath is set to `$PREFIX/lib` but I can't find how in meson docs to find where libraries are installed but clearly its different for different systems but this needs to be set to it (or the install set).

Reply:
Maybe `shared_library()` takes `install-dir:`?